hail/src/main/scala/is/hail/expr/types/physical/PCanonicalString.scala

```diff
@@ -73,6 +66,12 @@ abstract class PCanonicalString(val required: Boolean) extends PString {
       dstAddress
     )
   }
+
+  def constructAtAddress(mb: MethodBuilder, addr: Code[Long], region: Code[Region], srcPType: PType, srcAddress: Code[Long], forceDeep: Boolean): Code[Unit] =
+    fundamentalType.constructAtAddress(mb, addr, region, srcPType, srcAddress, forceDeep)
```

Comment:
srcPType needs to be srcPType.asInstanceOf[PString].fundamentalType here and below

Reply:
fixed. We definitely want to only be calling this on fundamental type, I'll PR that after the staged piece goes in.